tempodb/backend/cache/cache.go

```diff
@@ -25,57 +29,57 @@ func NewCache(nextReader backend.RawReader, nextWriter backend.RawWriter, cache
 	return rw, rw, nil
 }
 
-// List implements backend.Reader
+// List implements backend.RawReader
 func (r *readerWriter) List(ctx context.Context, keypath backend.KeyPath) ([]string, error) {
 	return r.nextReader.List(ctx, keypath)
 }
 
-// Read implements backend.Reader
-func (r *readerWriter) Read(ctx context.Context, name string, keypath backend.KeyPath) ([]byte, error) {
+// Read implements backend.RawReader
+func (r *readerWriter) Read(ctx context.Context, name string, keypath backend.KeyPath, shouldCache bool) (io.ReadCloser, int64, error) {
 	var k string
-	if shouldCache(name) {
+	if shouldCache {
 		k = key(keypath, name)
 		found, vals, _ := r.cache.Fetch(ctx, []string{k})
 		if len(found) > 0 {
```

Comment:
If this is a partial hit, shouldn't it go to the storage to fetch the misses?

Reply:
The Cache interfaces are adopted from Cortex, but in Tempo there is no partial hit at the moment - its either a hit or a miss.

So if `len(found) == 0` the function will fetch object from the backend (line 48)